shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -80,25 +80,25 @@ public void onRemoval(RemovalNotification<String, Gauge> removal) {
                 public void onRemoval(RemovalNotification<String, Meter> removal) {
                     Meter meter = removal.getValue();
                     if (removal.getCause() != RemovalCause.REPLACED) {
-                        Exceptions.checkNotNullOrEmpty(meter.getName(), "meter");
-                        metrics.remove(meter.getName());
-                        log.debug("Removed Meter: {}.", meter.getName());
+                        metrics.remove(meter.getId());
+                        log.debug("Removed Meter: {}.", meter.getId());
                     }
                 }
             }).
             build();
     }
 
     @Override
-    public void incCounterValue(String name, long delta) {
+    public void incCounterValue(String name, long delta, String... tags) {
         Exceptions.checkNotNullOrEmpty(name, "name");
         Preconditions.checkNotNull(delta);
-        String counterName = name + ".Counter";
+        String counterName = nameFromTags(name, tags) + ".Counter";
         try {
             Counter counter = countersCache.get(counterName, new Callable<Counter>() {
```

Comment:
This countersCache interface is very strange taking a callable like this. Perhaps we should look into ways to optimize this.

Reply:
This is the callback to be called when the counter is expired. Yes overall it is more feasible to revisit the caching mechanism with the new properties provided by Micrometer. The effort will probably be after this migration (as we need to ensure the migration itself runs smoothly)